core/commands/repo.go

```diff
@@ -272,96 +270,69 @@ type VerifyProgress struct {
 	Progress int
 }
 
-var repoVerifyCmd = &oldcmds.Command{
+var repoVerifyCmd = &cmds.Command{
 	Helptext: cmdkit.HelpText{
 		Tagline: "Verify all blocks in repo are not corrupted.",
 	},
-	Run: func(req oldcmds.Request, res oldcmds.Response) {
-		nd, err := req.InvocContext().GetNode()
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		nd, err := cmdenv.GetNode(env)
 		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
+			return err
 		}
 
-		out := make(chan interface{})
-		res.SetOutput((<-chan interface{})(out))
-		defer close(out)
-
 		bs := bstore.NewBlockstore(nd.Repo.Datastore())
 		bs.HashOnRead(true)
 
-		keys, err := bs.AllKeysChan(req.Context())
+		keys, err := bs.AllKeysChan(req.Context)
 		if err != nil {
 			log.Error(err)
-			return
+			return err
 		}
 
 		var fails int
 		var i int
 		for k := range keys {
 			_, err := bs.Get(k)
 			if err != nil {
-				select {
-				case out <- &VerifyProgress{
+				res.Emit(&VerifyProgress{
```

Comment:
Needs to check the error. Otherwise, we'll continue working even if the client has gone away.